Sources/Thumbprint/Components/Toast.swift

```diff
@@ -9,6 +9,76 @@ private protocol ToastViewDelegate: AnyObject {
  * A wrapper around ToastView to handle the toast presentation and dismissal animations
  */
 public class Toast: UIView {
+    public enum IconType: Equatable {
+        case defaultForTheme
+        case custom(Icon?)
+
+        func resolvedIcon(for theme: Toast.Theme) -> Icon? {
+            switch self {
+            case let .custom(icon):
+                return icon
+            case .defaultForTheme:
+                return theme.defaultIcon
+            }
+        }
+    }
+
+    public enum Theme {
```

Comment:
Although theme enum works as there wouldn't be much more customization needed. I like the approach of structs with static defaults ([example](https://github.com/thumbtack/ios/blob/2c36ea51f89ae6c36cc96e799752f4c30f1294df/ThumbprintUI/ThumbprintUI/Components/TPAlertBanner.swift#L3:L27)). It serves baseline purpose while enabling customization if needed. If you follow struct approach then we can incorporate icon within the struct and can get rid of `IconType` since anyone needed custom icon can initialize a new struct.

Reply:
According to design systems, the users shouldn't be able to customize the other properties (text color etc. ) . I'm still fine with using the struct for the improvements to the icon/theme if we can handle that.

Reply:
Updated to a struct. Thanks for the suggestion.